lib/src/embedded_cli.c

```diff
@@ -569,6 +588,30 @@ void embeddedCliTokenizeArgs(char *args) {
     args[insertPos + 1] = '\0';
 }
 
+static char *embeddedCliGetTokenVariable(char *tokenizedStr, uint8_t pos) {
+    if (tokenizedStr == NULL || pos == 0)
+        return NULL;
+    int i = 0;
+    int tokenCount = 1;
+    while (true) {
+        if (tokenCount == pos)
+            break;
+
+        if (tokenizedStr[i] == '\0') {
+            ++tokenCount;
+            if (tokenizedStr[i + 1] == '\0')
+                break;
+        }
+
+        ++i;
+    }
+
+    if (tokenizedStr[i] != '\0')
+        return &tokenizedStr[i];
+    else
+        return NULL;
+}
+
 const char *embeddedCliGetToken(const char *tokenizedStr, uint8_t pos) {
```

Comment:
Use created `embeddedCliGetTokenVariable` function instead of duplicating code.
Moreover, I declare all static functions at the top of the file with a small description on what it does, please use this approach. Notice that order of declaration is the same as order of definitions. And all definitions of static functions are placed after definitions of global functions (declared in embedded_cli.h)

Reply:
Heh, it wasn't super trivial to get all the const's to work right with zero warnings about discarding const qualifiers, so I did the ultra-lazy thing and duplicated code. I knew there must be a better way, but I just wanted to get something up for comments.

Reply:
Definitely don't need to change the name to cliBuffer32, but as for changing the type to an int32_t, I couldn't figure out a better way to do it while *never* casting from uint8_t to something that needs to be aligned to a larger boundary (and thus eliminating all warnings).  

I try to get everything to zero wanrings -- and now I know why, turning that cast warning off was a bad idea.